src/system/init.rs

```diff
@@ -0,0 +1,93 @@
+use log::*;
+use std::ffi::CString;
+use syslog::{BasicLogger, Facility, Formatter3164};
+
+use super::*;
+
+fn mount_vfs(source_name: &str, target_name: &str, fstype: &str) {
+    info!("Mounting {} as type {}", target_name, fstype);
+    unsafe {
+        // CString constructor ensures the trailing 0byte, which is required by libc::mount
+        let src_c_ctr = CString::new(source_name).unwrap();
+        let target_name_c_ctr = CString::new(target_name).unwrap();
+        let fstype_c_ctr = CString::new(fstype).unwrap();
+        let options_c_ctr = CString::new("").unwrap();
+
+        let ret = libc::mount(
+            src_c_ctr.as_ptr() as *const i8,
+            target_name_c_ctr.as_ptr() as *const i8,
+            fstype_c_ctr.as_ptr() as *const i8,
+            0,
+            options_c_ctr.as_ptr() as *const libc::c_void,
+        );
+
+        if ret < 0 {
+            error!("Failed to mount ({})", ret);
+            libc::perror(
+                String::from("Error: ").as_bytes().as_ptr() as *const i8
+            );
+        }
+    }
+}
+
+pub fn init_rootfs() {
+    if get_pid() != 1 {
+        warn!("Trying to initialize rootfs but auraed is not run as pid1. Abort setup of rootfs.");
+        return;
+    }
+
+    mount_vfs("none", "/dev", "devtmpfs");
+    mount_vfs("none", "/sys", "sysfs");
+    mount_vfs("proc", "/proc", "proc");
```

Comment:
This sounds a bit crazy, however I have a use case for a system where `procfs` is NOT mounted.

Is it possible to add a compile time flag or similar to run `auraed` in an extremely risky experimental mode where `procfs` is not mounted?

Reply:
Sure! Will add a simple and fast solution to this PR. Maybe it makes sense to have the mount points configurable? Opened an issue here to discuss and track https://github.com/aurae-runtime/auraed/issues/25

Reply:
Maybe a `noprocfs` kernel parameter?